client/state/initial-state.js

```diff
@@ -193,13 +193,16 @@ export function persistOnChange( reduxStore ) {
 			const serializedState = serialize( state, reduxStore.getCurrentReducer() );
 			const _timestamp = Date.now();
 
-			const storeTasks = map( serializedState.get(), ( data, storageKey ) =>
-				persistentStoreState( reduxStateKey, storageKey, data, _timestamp )
-			);
-
-			Promise.all( storeTasks ).catch( setError =>
-				debug( 'failed to set redux-store state', setError )
-			);
+			// clear out storage first to prevent safari's WAL from growing forever
+			clearStorage().then( () => {
```

Comment:
This won't work well with the current dynamically loaded reducers:
- first session: loads the `root` and the `signup` reducer, stores `root` and the `signup` state
- second session: loads the `root` and `woo` reducers, but not `signup` (not on a signup route any more), then clears storage and stores `root` and `woo` state. The `signup` state will be wiped out.

The series of patches by @sgomes will change that, as it always loads all state. But at this moment, let's think about whether we're not breaking signup -- that's the only section that I think is at risk.

Reply:
Good point. 

Maybe we break this out a different way. Load all of the current keys, clear the db, then save them all again. Though if we did that we'd also need a `isPurging` type state that prevented writes during the async clear process.

Reply:
Hmm, if we wanted to do that, landing https://github.com/Automattic/wp-calypso/pull/38320 would make it much easier, as that gives us a supported way of getting all the current values.

Reply:
> Load all of the current keys, clear the db, then save them all again.

Yes, that should work well. The race condition can be avoided by adding some promise-based equivalent of a mutex.

>  Hmm, if we wanted to do that, landing #38320 would make it much easier

I agree. I can review that work ASAP and see if it can be merged quickly. It would help a lot with this high-priority issue.